app/components/annotation/Article.js

```diff
@@ -9,7 +23,35 @@ export default React.createClass({
 
   propTypes: {
     article: React.PropTypes.object.isRequired,
-    topics: React.PropTypes.array.isRequired
+    topics: React.PropTypes.array.isRequired,
+    onHighlight: React.PropTypes.function,
+    highlights: React.PropTypes.array
+  },
+
+  handleClick: function() {
```

Comment:
it is - after some data munging. the code in this function cleans up the data nicely, then calls `onHighlight`, which causes a dispatch

i believe this is the right pattern for `react-redux` usage and it feels good to me. the only thing is the data munging might belong elsewhere but it's nitty gritty DOM stuff, which feels like it belongs down in the component next to the down, not raised up in an action or a reducer somewhere